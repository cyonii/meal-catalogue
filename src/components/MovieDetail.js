import { useEffect, useState } from 'react';
import { useParams } from 'react-router';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import MovieReviews from './MovieReviews';
import Loading from './widgets/Loading';
import Movie404 from './widgets/Movie404';
import { moviedb } from '../api/movies';

const MovieDetail = () => {
  const { id } = useParams();
  const [movie, setMovie] = useState({});
  const [loading, setLoading] = useState(true);
  const [notFound, setNotFound] = useState(false);

  useEffect(async () => {
    setLoading(true);
    setNotFound(false);
    await moviedb.movieInfo(id)
      .then((data) => setMovie(data))
      .catch((err) => {
        if (err.response.status === 404) setNotFound(true);
      });
    setLoading(false);
  }, []);

  // Return loading indicator if movie data is not fetched
  if (loading) return (<Loading />);

  // Return 404 if movie not found
  if (notFound) return (<Movie404 />);

  const makeBadge = (text) => <div className="badge bg-secondary lh-1 py-2 px-3 me-2 mb-2" key={text}>{text}</div>;
  const makeAttrColumn = (colClass, text, value) => (
    <Col className={colClass}>
      <div className="mb-3">
        <small className="d-block text-muted fw-bold">{`${text}:`}</small>
        {value instanceof Array ? value : <b>{value}</b>}
      </div>
    </Col>
  );

  return (
    <Container className="mt-5 text-white mb-5" data-testid="movie-detail">
      <Row>
        <Col md={5} lg={4}>
          <img className="img-fluid" id="poster" src={`https://image.tmdb.org/t/p/w780/${movie.poster_path}`} alt={movie.title} />
        </Col>
        <Col>
          <h1 className="text-secondarys lh-1 mb-0"><b>{movie.title}</b></h1>
          <a href={movie.homepage} className="text-secondary fw-light" target="_blank" rel="noreferrer">
            <small>{movie.homepage}</small>
          </a>

          <p className="lh-sm text-info mt-4">{movie.overview}</p>
          <Row>
            {makeAttrColumn('col-6 col-lg-4', 'STATUS', movie.status)}
            {makeAttrColumn('col-6 col-lg-4', 'RELEASE DATE', movie.release_date)}
            {makeAttrColumn('col-6 col-lg-4', 'RUNTIME', `${movie.runtime} Minutes`)}
            {makeAttrColumn('col-6 col-lg-4', 'RATING', `${movie.vote_average} / 10`)}
            {makeAttrColumn('col-6 col-lg-4', 'VOTE COUNT', `${movie.vote_count}`)}
            {makeAttrColumn('col-6 col-lg-4', 'POPULARITY', `${movie.popularity}%`)}
            <Row>
              {makeAttrColumn('col-12 col-md-6', 'GENRES', movie.genres.map((genre) => makeBadge(genre.name)))}
              {makeAttrColumn('col-12 col-md-6', 'SPOKEN LANGUAGES', movie.spoken_languages.map((lang) => makeBadge(lang.name)))}
            </Row>
          </Row>
        </Col>
      </Row>

      <hr className="mb-5" />

      <Row className="align-items-center row-type-1 overflow-hidden">
        <Col className="overflow-hidden overflow-y-auto h-100 mt-3 mt-md-0 pb-3">
          <MovieReviews movieID={movie.id} />
        </Col>
        <Col md={5} className="order-first order-md-last text-center">
          {movie.tagline ? <div className="alert alert-info border-0 border-bottom border-2 border-info rounded-0 py-2 mb-0 lh-1">{movie.tagline}</div> : ''}
          <img className="img-fluid" src={`https://image.tmdb.org/t/p/w780/${movie.backdrop_path}`} alt={movie.title} />
        </Col>
      </Row>
    </Container>
  );
};

export default MovieDetail;
