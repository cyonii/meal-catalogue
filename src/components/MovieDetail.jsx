import { useEffect, useState } from "react";
import { useParams } from "react-router";
import MovieReviews from "./MovieReviews";
import MovieCredits from "./MovieCredits";
import Loading from "./widgets/Loading";
import Movie404 from "./widgets/Movie404";
import { moviedb } from "../api/movies";

// Movie attributes component
const Attrbute = (props) => {
  const { label, value } = props;

  return (
    <div>
      <h6 className="text-slate-500 text-xs font-bold">{`${label}:`}</h6>
      {value instanceof Array ? (
        value.map((item) => (
          <span
            className="inline-block bg-slate-500 leading-none rounded py-1 px-3 mr-2 mb-2"
            key={item.name || item.english_name}
          >
            {item.name || item.english_name}
          </span>
        ))
      ) : (
        <b className="text-slate-300">{value}</b>
      )}
    </div>
  );
};

const MovieDetail = () => {
  const { id } = useParams();
  const [movie, setMovie] = useState({});
  const [loading, setLoading] = useState(true);
  const [notFound, setNotFound] = useState(false);
  const intComma = new Intl.NumberFormat().format;

  useEffect(async () => {
    setLoading(true);
    setNotFound(false);
    await moviedb
      .movieInfo(id)
      .then((data) => setMovie(data))
      .catch((err) => {
        if (err.response.status === 404) setNotFound(true);
      });
    setLoading(false);
  }, []);

  // Return loading indicator if movie data is not fetched
  if (loading) return <Loading />;

  // Return 404 if movie not found
  if (notFound) return <Movie404 />;

  return (
    <div className="container text-slate-50 py-5" data-testid="movie-detail">
      <div
        className="flex flex-col md:flex-row gap-4"
        style={{
          backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.9), rgba(0, 0, 0, 0.9)), url(https://image.tmdb.org/t/p/w780/${movie.backdrop_path})`,
          backgroundSize: "cover",
          backgroundRepeat: "no-repeat",
        }}
      >
        <div className="basis-1/3">
          <img
            className="img-fluid"
            id="poster"
            src={`https://image.tmdb.org/t/p/w780/${movie.poster_path}`}
            alt={movie.title}
          />
        </div>

        <div className="basis-2/3 p-3">
          <h1 className="text-4xl text-slate-50 leading-none">
            <b>{movie.title}</b>
          </h1>
          <a
            href={movie.homepage}
            className="text-slate-500 font-light text-lg"
            target="_blank"
            rel="noreferrer"
          >
            <small>{movie.homepage}</small>
          </a>
          {movie.tagline && (
            <div className="bg-cyan-100 text-cyan-900 p-2 border-b-2 border-cyan-900 leading-none">
              {movie.tagline}
            </div>
          )}
          <p className="leading-5 text-cyan-600 mt-4">{movie.overview}</p>

          <hr className="my-5 border-slate-900" />

          <div className="grid grid-cols-3 lg:grid-cols-4 gap-4 mt-5">
            <Attrbute label="STATUS" value={movie.status} />
            <Attrbute label="RELEASE DATE" value={movie.release_date} />
            <Attrbute label="RUNTIME" value={`${movie.runtime} Minutes`} />
            <Attrbute label="RATING" value={`${movie.vote_average} / 10`} />
            <Attrbute label="VOTE COUNT" value={`${movie.vote_count}`} />
            <Attrbute
              label="POPULARITY"
              value={`${intComma(movie.popularity)}%`}
            />
            <Attrbute label="BUDGET" value={`$${intComma(movie.budget)}`} />
            <Attrbute label="REVENUE" value={`$${intComma(movie.revenue)}`} />
          </div>

          <hr className="my-5 border-slate-900" />

          <div className="grid grid-cols-2 gap-4">
            <Attrbute label="GENRES" value={movie.genres} />
            <Attrbute label="SPOKEN LANGUAGES" value={movie.spoken_languages} />
          </div>
        </div>
      </div>

      {/* Movie reviews */}
      <hr className="my-8 md:my-10 border-slate-900" />
      <div className="flex flex-col md:flex-row h-[400px] gap-4">
        <div className="md:basis-1/2 overflow-hidden overflow-y-auto h-full mt-3 md:mt-0 pb-3">
          <MovieReviews movieID={movie.id} />
        </div>

        <div className="md:basis-1/2 order-first md:order-last text-center">
          <img
            className="img-fluid"
            src={`https://image.tmdb.org/t/p/w780/${movie.backdrop_path}`}
            alt={movie.title}
          />
        </div>
      </div>

      {/* Credits row */}
      <hr className="my-5 border-slate-900" />
      <MovieCredits movieID={movie.id} />
    </div>
  );
};

export default MovieDetail;
