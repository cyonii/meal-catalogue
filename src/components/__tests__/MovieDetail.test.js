import {
  screen, render, waitFor, cleanup,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MovieDetail from '../MovieDetail';
import singleMovieResponse from './data/singleMovie.json';
import reviewResponse from './data/reviews.json';

const mockedMovieResponse = singleMovieResponse;
const mockedReviewResponse = reviewResponse;

// mock moviedb api call for movieInfo and movieReviews
jest.mock('../../api/movies', () => ({
  moviedb: {
    movieInfo: () => Promise.resolve(mockedMovieResponse),
    movieReviews: () => Promise.resolve(mockedReviewResponse),
  },
}));

describe('MovieDetail', () => {
  afterEach(cleanup);

  it('should render Loading component while fetching movie', async () => {
    render(
      <MemoryRouter initialEntries={['/movie/497698']}>
        <MovieDetail />
      </MemoryRouter>,
    );

    await waitFor(() => {
      expect(screen.getByTestId('loading')).toBeInTheDocument();
      expect(screen.getByTestId('loading')).toMatchSnapshot();
    });
  });

  it('should render review list items', async () => {
    render(
      <MemoryRouter initialEntries={['/movie/497698']}>
        <MovieDetail />
      </MemoryRouter>,
    );

    await waitFor(() => {
      expect(screen.getByTestId('movie-detail')).toMatchSnapshot();
      expect(screen.getByTestId('review-list')).toBeInTheDocument();
      expect(screen.getAllByTestId('review-item').length).toBe(3);
    });
  });
});
