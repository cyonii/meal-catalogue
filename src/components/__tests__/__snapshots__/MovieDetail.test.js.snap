// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail should render Loading component while fetching movie 1`] = `
<div
  class="d-flex flex-column justify-content-center align-items-center mt-5"
  data-testid="loading"
>
  <i
    class="bi bi-lightning-charge text-secondary spin"
    style="font-size: 1.5rem;"
  />
</div>
`;

exports[`MovieDetail should render review list items 1`] = `
<div
  class="mt-5 text-white mb-5 container"
  data-testid="movie-detail"
>
  <div
    class="row"
  >
    <div
      class="col-lg-4 col-md-5"
    >
      <img
        alt="Fight Club"
        class="img-fluid"
        id="poster"
        src="https://image.tmdb.org/t/p/w780/null"
      />
    </div>
    <div
      class="col"
    >
      <h1
        class="text-secondarys lh-1 mb-0"
      >
        <b>
          Fight Club
        </b>
      </h1>
      <a
        class="text-secondary fw-light"
        href=""
        rel="noreferrer"
        target="_blank"
      >
        <small />
      </a>
      <p
        class="lh-sm text-info mt-4"
      >
        A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground "fight clubs" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.
      </p>
      <div
        class="row"
      >
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              STATUS:
            </small>
            <b>
              Released
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RELEASE DATE:
            </small>
            <b>
              1999-10-12
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RUNTIME:
            </small>
            <b>
              139 Minutes
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RATING:
            </small>
            <b>
              7.8 / 10
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              VOTE COUNT:
            </small>
            <b>
              3439
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              POPULARITY:
            </small>
            <b>
              0.5%
            </b>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-md-6 col"
          >
            <div
              class="mb-3"
            >
              <small
                class="d-block text-muted fw-bold"
              >
                GENRES:
              </small>
              <div
                class="badge bg-secondary lh-1 py-2 px-3 me-2 mb-2"
              >
                Drama
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-6 col"
          >
            <div
              class="mb-3"
            >
              <small
                class="d-block text-muted fw-bold"
              >
                SPOKEN LANGUAGES:
              </small>
              <div
                class="badge bg-secondary lh-1 py-2 px-3 me-2 mb-2"
              >
                English
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="mb-5"
  />
  <div
    class="align-items-center row-type-1 overflow-hidden row"
  >
    <div
      class="overflow-hidden overflow-y-auto h-100 mt-3 mt-md-0 pb-3 col"
    >
      <h5
        class="sticky-top bg-dark py-2 text-decoration-underline border-secondary"
      >
        Reviews
      </h5>
      <ul
        class="list-group list-group-flush"
        data-testid="review-list"
      >
        <h2
          class="my-auto opacity-25 fw-bold mb-0 lh-1"
        >
          No reviews yet.
        </h2>
      </ul>
    </div>
    <div
      class="order-first order-md-last text-center col-md-5"
    >
      <div
        class="alert alert-info border-0 border-bottom border-2 border-info rounded-0 py-2 mb-0 lh-1"
      >
        How much can you know about yourself if you've never been in a fight?
      </div>
      <img
        alt="Fight Club"
        class="img-fluid"
        src="https://image.tmdb.org/t/p/w780//fCayJrkfRaCRCTh8GqN30f8oyQF.jpg"
      />
    </div>
  </div>
</div>
`;

exports[`MovieDetail should render review list items 2`] = `
<div
  class="mt-5 text-white mb-5 container"
  data-testid="movie-detail"
>
  <div
    class="row"
  >
    <div
      class="col-lg-4 col-md-5"
    >
      <img
        alt="Fight Club"
        class="img-fluid"
        id="poster"
        src="https://image.tmdb.org/t/p/w780/null"
      />
    </div>
    <div
      class="col"
    >
      <h1
        class="text-secondarys lh-1 mb-0"
      >
        <b>
          Fight Club
        </b>
      </h1>
      <a
        class="text-secondary fw-light"
        href=""
        rel="noreferrer"
        target="_blank"
      >
        <small />
      </a>
      <p
        class="lh-sm text-info mt-4"
      >
        A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground "fight clubs" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.
      </p>
      <div
        class="row"
      >
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              STATUS:
            </small>
            <b>
              Released
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RELEASE DATE:
            </small>
            <b>
              1999-10-12
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RUNTIME:
            </small>
            <b>
              139 Minutes
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RATING:
            </small>
            <b>
              7.8 / 10
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              VOTE COUNT:
            </small>
            <b>
              3439
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              POPULARITY:
            </small>
            <b>
              0.5%
            </b>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-md-6 col"
          >
            <div
              class="mb-3"
            >
              <small
                class="d-block text-muted fw-bold"
              >
                GENRES:
              </small>
              <div
                class="badge bg-secondary lh-1 py-2 px-3 me-2 mb-2"
              >
                Drama
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-6 col"
          >
            <div
              class="mb-3"
            >
              <small
                class="d-block text-muted fw-bold"
              >
                SPOKEN LANGUAGES:
              </small>
              <div
                class="badge bg-secondary lh-1 py-2 px-3 me-2 mb-2"
              >
                English
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="mb-5"
  />
  <div
    class="align-items-center row-type-1 overflow-hidden row"
  >
    <div
      class="overflow-hidden overflow-y-auto h-100 mt-3 mt-md-0 pb-3 col"
    >
      <h5
        class="sticky-top bg-dark py-2 text-decoration-underline border-secondary"
      >
        Reviews
      </h5>
      <div
        class="d-flex flex-column justify-content-center align-items-center mt-5"
        data-testid="loading"
      >
        <i
          class="bi bi-lightning-charge text-secondary spin"
          style="font-size: 1rem;"
        />
      </div>
    </div>
    <div
      class="order-first order-md-last text-center col-md-5"
    >
      <div
        class="alert alert-info border-0 border-bottom border-2 border-info rounded-0 py-2 mb-0 lh-1"
      >
        How much can you know about yourself if you've never been in a fight?
      </div>
      <img
        alt="Fight Club"
        class="img-fluid"
        src="https://image.tmdb.org/t/p/w780//fCayJrkfRaCRCTh8GqN30f8oyQF.jpg"
      />
    </div>
  </div>
</div>
`;

exports[`MovieDetail should render review list items 3`] = `
<div
  class="mt-5 text-white mb-5 container"
  data-testid="movie-detail"
>
  <div
    class="row"
  >
    <div
      class="col-lg-4 col-md-5"
    >
      <img
        alt="Fight Club"
        class="img-fluid"
        id="poster"
        src="https://image.tmdb.org/t/p/w780/null"
      />
    </div>
    <div
      class="col"
    >
      <h1
        class="text-secondarys lh-1 mb-0"
      >
        <b>
          Fight Club
        </b>
      </h1>
      <a
        class="text-secondary fw-light"
        href=""
        rel="noreferrer"
        target="_blank"
      >
        <small />
      </a>
      <p
        class="lh-sm text-info mt-4"
      >
        A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground "fight clubs" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.
      </p>
      <div
        class="row"
      >
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              STATUS:
            </small>
            <b>
              Released
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RELEASE DATE:
            </small>
            <b>
              1999-10-12
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RUNTIME:
            </small>
            <b>
              139 Minutes
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              RATING:
            </small>
            <b>
              7.8 / 10
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              VOTE COUNT:
            </small>
            <b>
              3439
            </b>
          </div>
        </div>
        <div
          class="col-6 col-lg-4 col"
        >
          <div
            class="mb-3"
          >
            <small
              class="d-block text-muted fw-bold"
            >
              POPULARITY:
            </small>
            <b>
              0.5%
            </b>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-md-6 col"
          >
            <div
              class="mb-3"
            >
              <small
                class="d-block text-muted fw-bold"
              >
                GENRES:
              </small>
              <div
                class="badge bg-secondary lh-1 py-2 px-3 me-2 mb-2"
              >
                Drama
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-6 col"
          >
            <div
              class="mb-3"
            >
              <small
                class="d-block text-muted fw-bold"
              >
                SPOKEN LANGUAGES:
              </small>
              <div
                class="badge bg-secondary lh-1 py-2 px-3 me-2 mb-2"
              >
                English
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="mb-5"
  />
  <div
    class="align-items-center row-type-1 overflow-hidden row"
  >
    <div
      class="overflow-hidden overflow-y-auto h-100 mt-3 mt-md-0 pb-3 col"
    >
      <h5
        class="sticky-top bg-dark py-2 text-decoration-underline border-secondary"
      >
        Reviews
      </h5>
      <ul
        class="list-group list-group-flush"
        data-testid="review-list"
      >
        <li
          class="px-0 bg-transparent text-white border-0 border-bottom border-secondary list-group-item"
          data-testid="review-item"
        >
          <h6
            class="text-info mb-0 lh-1"
          >
            Cat Ellington
          </h6>
          <div
            class="d-flex justify-content-between align-items-center mb-3"
          >
            <small
              class="text-muted"
            >
              Rating: null
            </small>
          </div>
          <p
            class="mb-2 opacity-75 review-text"
          >
            (As I'm writing this review, Darth Vader's theme music begins to build in my mind...)
          </p>
          <p
            class="mb-2 opacity-75 review-text"
          >
            
          </p>
          <p
            class="mb-2 opacity-75 review-text"
          >
            Well, it actually has a title, what the Darth Vader theme. And that title is "The Imperial March", composed by the great John Williams, whom, as many of you may 
            <span
              class="text-muted fw-bold"
            >
              ...
            </span>
          </p>
          <div
            class="d-flex mt-1 opacity-100"
          >
            <button
              class="lh-1 py-0 pb-1 rounded-0 text-decoration-none btn btn-info"
              type="button"
            >
              <small
                class="fw-bold"
                style="font-size: 13px;"
              >
                Show more
              </small>
            </button>
          </div>
        </li>
        <li
          class="px-0 bg-transparent text-white border-0 border-bottom border-secondary list-group-item"
          data-testid="review-item"
        >
          <h6
            class="text-info mb-0 lh-1"
          >
            John Chard
          </h6>
          <div
            class="d-flex justify-content-between align-items-center mb-3"
          >
            <small
              class="text-muted"
            >
              Rating: 10
            </small>
          </div>
          <p
            class="mb-2 opacity-75 review-text"
          >
            A long time ago in a childhood not too far away...
          </p>
          <p
            class="mb-2 opacity-75 review-text"
          >
            
          </p>
          <p
            class="mb-2 opacity-75 review-text"
          >
            Princess Leia is captured and held hostage by the Imperial Army as it seeks to rule the Galactic Empire. An old Jedi Knight by the name of Ben "Obi-Wan" Kenobi may just be hers, and the rebels onl
            <span
              class="text-muted fw-bold"
            >
              ...
            </span>
          </p>
          <div
            class="d-flex mt-1 opacity-100"
          >
            <button
              class="lh-1 py-0 pb-1 rounded-0 text-decoration-none btn btn-info"
              type="button"
            >
              <small
                class="fw-bold"
                style="font-size: 13px;"
              >
                Show more
              </small>
            </button>
          </div>
        </li>
        <li
          class="px-0 bg-transparent text-white border-0 border-bottom border-secondary list-group-item"
          data-testid="review-item"
        >
          <h6
            class="text-info mb-0 lh-1"
          >
            gastyny
          </h6>
          <div
            class="d-flex justify-content-between align-items-center mb-3"
          >
            <small
              class="text-muted"
            >
              Rating: 10
            </small>
          </div>
          <p
            class="mb-2 opacity-75 review-text"
          >
            Star Wars (1977) is a true masterpiece of cinema, and is 
          </p>
          <p
            class="mb-2 opacity-75 review-text"
          >
              definitely one of the best films ever made.
          </p>
          <p
            class="mb-2 opacity-75 review-text"
          >
            
          </p>
          <p
            class="mb-2 opacity-75 review-text"
          >
            For me Star Wars (1977) is the best movie of all time,tied with The Lord of the Rings trilogy. Star Wars (1977) it is for sure the most iconic
            <span
              class="text-muted fw-bold"
            >
              ...
            </span>
          </p>
          <div
            class="d-flex mt-1 opacity-100"
          >
            <button
              class="lh-1 py-0 pb-1 rounded-0 text-decoration-none btn btn-info"
              type="button"
            >
              <small
                class="fw-bold"
                style="font-size: 13px;"
              >
                Show more
              </small>
            </button>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="order-first order-md-last text-center col-md-5"
    >
      <div
        class="alert alert-info border-0 border-bottom border-2 border-info rounded-0 py-2 mb-0 lh-1"
      >
        How much can you know about yourself if you've never been in a fight?
      </div>
      <img
        alt="Fight Club"
        class="img-fluid"
        src="https://image.tmdb.org/t/p/w780//fCayJrkfRaCRCTh8GqN30f8oyQF.jpg"
      />
    </div>
  </div>
</div>
`;
